### PR TITLE
Add input validation

### PR DESCRIPTION
**Pull request by caio-galvao**
Validates the input files.

### CODE REVIEW

File: build-simulation.py [hunk 1]
```diff
@@ -75,11 +126,10 @@ def generate_list(values, t, num_instances, num_markets):
 raw_sp_input = pd.read_csv('data/input_sp.csv')
 instances = raw_input['instance'].value_counts()
 
-if checkInputSP(raw_sp_input, instances) == False:
-    raise ValueError("Error in the savings plan input.")
-
 raw_demand = pd.read_csv('data/TOTAL_demand.csv')
 
+validate_input(raw_input, raw_sp_input, raw_demand)
```
Comment (thiagomanel) [inline build-simulation.py:131]: uhm ...
i think the code would be better if we have three functions here:
1. validate_raw_input
2. validate_sp_input (what is sp btw? it is not clear)
3. validate_demand

Comment (thiagomanel) [reply build-simulation.py:131]: of course, each one of these methods will call their own inner different functions

**Review by thiagomanel — CHANGES_REQUESTED** [review]
I think we need to reorganize a little bit

File: build-simulation.py (hunk 1 above)
Comment (caio-galvao) [reply build-simulation.py:131]: Ok.

Comment (caio-galvao) [reply build-simulation.py:131]: sp means savings plan, i will add this in the doc.

File: build-simulation.py [hunk 2]
```diff
@@ -30,10 +30,13 @@ def main():
     raw_sp_input = pd.read_csv(sys.argv[2])
     raw_demand = pd.read_csv(sys.argv[3])
 
-    instances = raw_input['instance'].value_counts()
-
-    if checkInputSP(raw_sp_input, instances) == False:
-        raise ValueError("Error in the savings plan input.")
+    validate_input(raw_input)
```
Comment (thiagomanel) [inline build-simulation.py:33]: update code and documentation to rename *_input to $name_of_market_input

**Review by thiagomanel — CHANGES_REQUESTED** [review]
mostly renaming stuff

**Review by thiagomanel — APPROVED** [review]
lgtm

**Pull request merged**